### ISSUE TITLE
Got "Error happend while scanning context"

### ISSUE THREAD

**Issue by Hugonee** [mebjas/html5-qrcode#529]
Hello, your barcode reader is the most powerful I seen!

I am trying to use it without the UI that you provide, but no matter what I try I got error message: "Error happend while scanning context"

here is the simple code:

<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<script src="https://unpkg.com/html5-qrcode" type="text/javascript"></script>

<body>
    <div id="qr-reader" style="width:450px">
        <div id="qr-reader-results" style="width:350px"></div>
    </div>
    

</body>

<script>

    // This method will trigger user permissions
    Html5Qrcode.getCameras().then(devices => {
        /**
         * devices would be an array of objects of type:
         * { id: "id", label: "label" }
         */
        if (devices && devices.length) {
            var cameraId = devices[0].id;
            const choosenCamera = { facingMode: { exact: "environment" }};
            const html5QrCode = new Html5Qrcode('qr-reader', /* verbose= */ true);
            const config = { fps: 1, qrbox: { width: 250, height: 250 } };
            html5QrCode.start(choosenCamera, config,
                (decodedText, decodedResult) => {
                    console.log({decodedText, decodedResult})
                    document.getElementById('qr-reader-results').innerText = decodedText;
                    html5QrCode.clear();
                },
                (errorMessage) => {
                    console.log({errorMessage})
                    document.getElementById('qr-reader-results').innerText = errorMessage;
                    html5QrCode.clear();
                })
                .catch((err) => {
                    console.log({err})
                    document.getElementById('qr-reader-results').innerText = err;
                    html5QrCode.clear();
                });

        }
    }).catch(err => {
        document.getElementById('qr-reader-results').innerText = err;
        // handle err
    });

</script>

</html>

**Comment by ROBERT-MCDOWELL** [comment]
did you try various qr code? should be black on white.

**Comment by Hugonee** [comment]
> did you try various qr code? should be black on white.

Its giving the error continuosly even without pointing to any QR

**Comment by ROBERT-MCDOWELL** [comment]
you have a camera problem I think

**Comment by Hugonee** [comment]
> you have a camera problem I think

I had tried with different devices iOS and Android Versions (7 to 10) and same error occurs

**Comment by ROBERT-MCDOWELL** [comment]
your code is incomplete.... check twice how to create an html5qrcode object.

**Comment by Hugonee** [comment]
Thanks cor tour support!

is there anything else than
`let html5QrCode = new Html5Qrcode(
	"reader", { fps: 10, qrbox: 250 } /* verbose= */ true); `

in order to create a  Html5Qrcode?

thanks in advance !!

**Comment by ROBERT-MCDOWELL** [comment]
yes, more, check the docs and read it, then search on issues list about smartphone etc...

**Comment by Hugonee** [comment]
I read the docs, reading as careful as I can, there is not a complete example without the JavaScript that got the UI.

Anyway, if you already catch what I am doing badly regarding the creation of the object, I really would appreciate your support showing me. Sometimes (as is the case) a person try and repeat and don’t find the point.

thanks again for your time, will be great if you could correct me!

**Comment by ROBERT-MCDOWELL** [comment]
there is no support here, only developers and users to fix bugs. yours is not a bug but code misunderstanding.
I suggest you try first the htmlqrcodescanner code before to post issues and see if it works for you.

**Comment by Hugonee** [comment]
#

> there is no support here, only developers and users to fix bugs. yours is not a bug but code misunderstanding.
> I suggest you try first the htmlqrcodescanner code before to post issues and see if it works for you.

I know, I tried the code, as you can see I congratulate the author because is the best QR and Barcode by far… 
There is two versions for the scanner one with UI that works just perfect, the other without UI is the one I am trying to use and with the same code does not work.
anyway I think more people could find same problem as me, would be good to have the basic code example for the non UI version, and that is what I would do when I fix it.
Would you help me on that?
Thank you in advance

**Comment by ROBERT-MCDOWELL** [comment]
no.

**Comment by Hugonee** [comment]
> no.

Well is a pity that you didn’t realized it’s not same API (that you speak about) and feel free to explain how GitHub works… it’s better to help the community than to pretend to be a master,
Thanks anyway for your time, please do not fill anymore with comments, I would like someone to help me and don’t need to make people boring with “master lessons” senselessness.

thanks!

**Comment by liushengqi000** [comment]
emmmm....
Error information can be removed by "new Html5Qrcode('qr-reader', false);"
"new Html5Qrcode(text,config:object,debugger:bool)" or "new Html5Qrcode(text,debugger:bool) "
you have to added "html5QrCode.stop(); " before "html5QrCode.clear();"

html5-qrcode Code
`
constructor(t, e) {"boolean" == typeof e ? this.verbose = !0 === e : e && this.verbose = !0 === e.verbose}
catch (t) {this.verbose && console.log(t)}
`

try 
` Html5Qrcode.getCameras().then(devices => {
 	if (devices && devices.length) {
 		var cameraId = devices[0].id;
 		const choosenCamera = {
 			facingMode: "user"
 		};
 		const html5QrCode = new Html5Qrcode('qr-reader', false);
 		const config = {
 			fps: 1,
 			qrbox: {
 				width: 250,
 				height: 250
 			}
 		};
 		html5QrCode.start(choosenCamera, config, (decodedText, decodedResult) => {
 			console.log({
 				decodedText, decodedResult
 			});
 			document.getElementById('qr-reader-results').innerText = decodedText;
 			html5QrCode.stop();
 			html5QrCode.clear();
 		}, (errorMessage) => {
 			console.log({
 				errorMessage
 			});
 			document.getElementById('qr-reader-results').innerText = errorMessage;
 		})
 		.catch ((err) => {
 			console.dir({
 				err
 			});
 			document.getElementById('qr-reader-results').innerText = err;
 			html5QrCode.stop();
 			html5QrCode.clear();
 		});
 	}
 }).catch (err => {
 	document.getElementById('qr-reader-results').innerText = err;
 });`

I hope that the next question can provide a text version of the code. I don't want to check the orc results

**Comment by liushengqi000** [comment]
> > did you try various qr code? should be black on white.
> 
> Its giving the error continuosly even without pointing to any QR

While verbose = true, it will keep telling you that there is no barcode found or no suitable decoder.
as you know,fps=1

**Comment by trasherdk** [comment]
` ```html `
formated code
` ``` `

```html
<head> 
<meta charset="UTF-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Document</title>
</head>
<script src="https://unpkg.com/html5-qrcode" type="text/javascript"></script>
<body>
<div id="qr-reader" style="width:450px"></div><div id="qr-reader-results" style="width:350px"></div>
</body> 
<script>Html5Qrcode.getCameras().then(devices => {
 	if (devices && devices.length) {
 		var cameraId = devices[0].id;
 		const choosenCamera = {
 			facingMode: {exact: "environment"}
 		};

 		const html5QrCode = new Html5Qrcode('qr-reader', false);
 		const config = {
 			fps: 1,
 			qrbox: {
 				width: 250,
 				height: 250
 			}
 		};

 		html5QrCode.start(choosenCamera, config, (decodedText, decodedResult) => {
 			console.log({
 				decodedText, decodedResult
 			});
 			document.getElementById('qr-reader-results').innerText = decodedText;
 			html5QrCode.stop();
 			html5QrCode.clear();
 		}, (errorMessage) => {
 			console.log({
 				errorMessage
 			});
 			document.getElementById('qr-reader-results').innerText = errorMessage;
 		})
 		.catch ((err) => {
 			console.dir({
 				err
 			});
 			document.getElementById('qr-reader-results').innerText = err;
 			html5QrCode.stop();
 			html5QrCode.clear();
 		});
 	}
 }).catch (err => {
 	document.getElementById('qr-reader-results').innerText = err;
 	html5QrCode.stop();
 	html5QrCode.clear();
 });
 </script>
```

**Comment by dbareis** [comment]
Was this ever resolved? If so how?
